### DISCUSSION TITLE
RawBlock html table: is it possible to use "special" cross reference syntax?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#8068] [Q&A]
### Description

Hi,
in this [sample page](https://ondata.github.io/linee-guida-opendata/small_table.html) I have added [`<span data-qmd="[-@sec-soggetti-destinatari]"></span>`](https://github.com/aborruso/linee-guida-opendata/blob/main/small_table.qmd#L16), to hide cross reference label and show only its ID.

But I have always the CR label.

Is it possible to use this special [CR syntax](https://quarto.org/docs/books/book-crossrefs.html#creating-references) in RawBlock html table? Did I make any syntax errors??

Thank you

**Comment by mcanouil** [comment]
Could you not hide links behind code contents and preferably share code snippets directly inside the post itself?

This is for two reasons:
- it's harder to read when having to go back and forth between the post and the links targets
- the text used to hide links has no meaning with the target links which means people with disabilities using screen readers, etc. will be completely lost and won't understand anything from your post. When using text as links, the text should be meaningful and let readers know what the link (and its content) is about.

**Reply by aborruso** [reply]
Thank you @mcanouil What I sent is just a silly example. In the actual example I will produce a rendering of the text that is understandable.

Is it normal that those syntax does not work?

**Reply by cscheid** [reply]
Can you provide a minimal, self-contained reproducible example so that we can follow up?

**Reply by aborruso** [reply]
> Can you provide a minimal, self-contained reproducible example so that we can follow up?

Thank you @cscheid. First of all, congratulations on your talk at posit-conf, very very interesting.

The small example:

- rendered https://ondata.github.io/linee-guida-opendata/debug/raw-html-cr.html
- the code https://github.com/ondata/linee-guida-opendata/blob/main/debug/raw-html-cr.qmd 

Actually the problem doesn't seem related to the HTML table, but it seems not to work on this page.
And there is something strange, which I see creating this example for you.

I have inserted this code

```
Please [see par. @sec-soggetti-destinatari]

Please [see par. -@sec-soggetti-destinatari]
```

and in both cases now I have only the CR ID.
It is as if there is a "broken" cache. How can I eleminate it and launch a clean render?

Thank you

![image](https://github.com/quarto-dev/quarto-cli/assets/30607/430f804e-ec6c-466f-b981-375779d2fc82)

**Reply by cscheid** [reply]
Thanks! However, we do need a "minimal, self-contained example". You have a whole book with a lot of content in this. Can you provide a _small_ example instead?

**Reply by aborruso** [reply]
> Can you provide a minimal, self-contained reproducible example so that we can follow up?

I'm confusing myself and closing because I can actually get what I needed. Excuse me.

But I have a documentation related question. In [this page](https://quarto.org/docs/books/book-crossrefs.html#creating-references) we have

> To refer to a chapter or appendix explicitly you should spell out “Chapter” or “Appendix” and use the number-only form of cross reference

But if I write

```
See [lorem -@sec-soggetti-destinatari]

See [lorem @sec-soggetti-destinatari]
```

I get in both cases only the ID. I get only "See lorem 1.1". So it seems that to get just the ID, just write anything before the CR. Whether there is `-` at the beginning of CR, or not, seems indifferent.

The small example https://aborruso.quarto.pub/sample/cr.html (the [code](https://github.com/aborruso/sample/blob/main/cr.qmd)).

Thank you

**Reply by cderv** [reply]
> I get in both cases only the ID. I get only "See lorem 1.1". So it seems that to get just the ID, just write anything before the CR. Whether there is - at the beginning of CR, or not, seems indifferent.

It seems so. From our codebase, we do use the Prefix passed when there is one, no matter about the `-` syntax, which is about suppressing author information in citeproc syntax. 

https://github.com/quarto-dev/quarto-cli/blob/476da182e527c3d5bd017cfcd7620f7f329e32b3/src/resources/filters/crossref/refs.lua#L47-L77

Not sure this has changed recently - maybe documentation was not updated 🤔
